packages/expo-updates/android/src/androidTest/java/expo/modules/updates/loader/RemoteLoaderTest.java

```diff
@@ -0,0 +1,267 @@
+package expo.modules.updates.loader;
+
+import android.content.Context;
+import android.net.Uri;
+
+import androidx.room.Room;
+import androidx.test.internal.runner.junit4.AndroidJUnit4ClassRunner;
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.stubbing.Answer;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+
+import expo.modules.manifests.core.LegacyManifest;
+import expo.modules.updates.UpdatesConfiguration;
+import expo.modules.updates.db.UpdatesDatabase;
+import expo.modules.updates.db.entity.AssetEntity;
+import expo.modules.updates.db.entity.UpdateEntity;
+import expo.modules.updates.db.enums.UpdateStatus;
+import expo.modules.updates.manifest.LegacyUpdateManifest;
+import expo.modules.updates.manifest.UpdateManifest;
+
+import static org.mockito.Mockito.*;
+
+@RunWith(AndroidJUnit4ClassRunner.class)
+public class RemoteLoaderTest {
```

Comment:
Are the Remote and Embedded Loader sub classes so distinct that these tests can't be unified into a `LoaderTest.java` file?